pkg/agent/wireguard/client_linux.go

```diff
@@ -66,13 +67,15 @@ type client struct {
 	wgClient                wgctrlClient
 	nodeName                string
 	k8sClient               clientset.Interface
+	mcClient                mcclientset.Interface
```

Comment:
`mcClient` and `mcNamespace` are used only in `Init()`. Could we change them to arguments of `Init()`? Even better, if we can change `k8sClient` to an argument of `Init()` too.

Reply:
Done. Move `k8sClient`, `mcClient` and `mcNamespace` to `Init()` now.